### PR TITLE
Bugfix: Answers to OStatus posts should never reach Diaspora

### PR DESCRIPTION
**Pull request by annando**
This was thought to be working - but it had issues. Now it should work greatly.

### CODE REVIEW

**Comment by Hypolite** [comment]
The word you're looking for is "bigly" 😉

File: include/notifier.php [hunk 1]
```diff
@@ -570,17 +568,20 @@ function notifier_run(&$argv, &$argc){
...
+		$r1 = array();
 
-		$r1 = q("SELECT DISTINCT(`batch`), `id`, `name`,`network` FROM `contact` WHERE `network` = '%s'
-			AND `uid` = %d AND `rel` != %d AND NOT `blocked` AND NOT `pending` AND NOT `archive` GROUP BY `batch` ORDER BY rand()",
-			dbesc(NETWORK_DIASPORA),
-			intval($owner['uid']),
-			intval(CONTACT_IS_SHARING)
-		);
+		if ($diaspora_delivery) {
+			if (!$followup)
```
Comment (Hypolite) [inline include/notifier.php]: Standards: Please add braces.

File: include/notifier.php [hunk 2]
```diff
@@ -479,11 +478,10 @@ function notifier_run(&$argv, &$argc){
 	if ($relocate)
```
Comment (Hypolite) [inline include/notifier.php]: Standards: Please add braces to this conditional statement.

File: include/notifier.php [hunk 3]
```diff
@@ -413,7 +418,7 @@ function notifier_run(&$argv, &$argc){
 			}
 
 			// Send a salmon to the parent owner
-			$r = q("SELECT `notify` FROM `contact` WHERE `nurl`='%s' AND `uid` IN (0, %d) AND `notify` != ''",
+			$r = q("SELECT `url`, `notify` FROM `contact` WHERE `nurl`='%s' AND `uid` IN (0, %d) AND `notify` != ''",
 				dbesc(normalise_link($thr_parent[0]['owner-link'])),
 				intval($uid));
 			if ($r)
```
Comment (Hypolite) [inline include/notifier.php]: Standards: Please use `dbm::is_result()` and add braces to this conditional statement.

File: include/notifier.php [hunk 4]
```diff
@@ -443,12 +448,6 @@ function notifier_run(&$argv, &$argc){
 			$sql_extra = " AND `network` IN ('".NETWORK_OSTATUS."', '".NETWORK_DFRN."')";
 		} else
```
Comment (Hypolite) [inline include/notifier.php]: Standards: Please add braces to this conditional statement.

File: include/notifier.php [hunk 5]
```diff
@@ -443,12 +448,6 @@ function notifier_run(&$argv, &$argc){
...
 			$sql_extra = " AND `network` IN ('".NETWORK_OSTATUS."', '".NETWORK_DFRN."', '".NETWORK_DIASPORA."', '".NETWORK_MAIL."', '".NETWORK_MAIL2."')";
-
-		$r = q("SELECT * FROM `contact` WHERE `id` IN ($conversant_str) AND NOT `blocked` AND NOT `pending` AND NOT `archive`".$sql_extra);
-
-		if (dbm::is_result($r))
-			$contacts = $r;
-
 	} else
```
Comment (Hypolite) [inline include/notifier.php]: Standards: Please add braces to this conditional statement.

File: include/notifier.php [hunk 6]
```diff
@@ -479,11 +478,10 @@ function notifier_run(&$argv, &$argc){
 	if ($relocate)
 		$r = $recipients_relocate;
 	else
```
Comment (Hypolite) [inline include/notifier.php]: Standards: Please add braces to this conditional statement.

**Comment by annando** [comment]
@Hypolite done.

File: include/notifier.php [hunk 7]
```diff
@@ -184,18 +185,18 @@ function notifier_run(&$argv, &$argc){
...
+		if (! count($items)) {
 			return;
 		}
 
 		// avoid race condition with deleting entries
 
-		if($items[0]['deleted']) {
-			foreach($items as $item)
+		if ($items[0]['deleted']) {
+			foreach ($items as $item)
```
Comment (Hypolite) [inline include/notifier.php]: Sorry I didn't catch that one, but braces seem missing from this `foreach` statement.

File: include/notifier.php [hunk 8]
```diff
@@ -326,7 +327,7 @@ function notifier_run(&$argv, &$argc){
 						dbesc(NETWORK_DFRN)
 					);
 					if (dbm::is_result($r))
```
Comment (Hypolite) [inline include/notifier.php]: Can you please add braces to the `if` and subsequent `foreach`?

File: include/notifier.php [hunk 9]
```diff
@@ -326,7 +327,7 @@ function notifier_run(&$argv, &$argc){
...
-						foreach($r as $rr)
+						foreach ($r as $rr)
```
Comment (Hypolite) [inline include/notifier.php]: Can you please add braces to this `foreach` statement?

**Comment by Hypolite** [comment]
@annando You're a doll ❤️

**Comment by annando** [comment]
@Hypolite hopefully this means something positive :)

**Comment by Hypolite** [comment]
Yes it does 😊

Equivalents: 
- "You're so sweet"
- "You're adorable"
- "You're a gem"

I'm not afraid to teach you both standards and english! 😎

**Pull request merged**